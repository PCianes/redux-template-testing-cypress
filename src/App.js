import React from 'react';
import styled from 'styled-components'
import logo from './logo.svg';
import { Counter } from './features/counter/Counter';
import './App.css';

function App() {
  const Title = styled.h1`
    background-color: rgb(112, 76, 182, .9);
    color: #fff;
    padding: 5px 15px;
    border-radius: 10px;
  `

  return (
    <div className="App">
      <header className="App-header">
        <Title>Redux template: Jest & Cypress</Title>
        <img src={logo} className="App-logo" alt="logo" />
        <Counter />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <span>
          <span>Learn </span>
          <a
            className="App-link"
            href="https://reactjs.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            React
          </a>
          <span>, </span>
          <a
            className="App-link"
            href="https://redux.js.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Redux
          </a>
          <span>, </span>
          <a
            className="App-link"
            href="https://redux-toolkit.js.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Redux Toolkit
          </a>
          ,<span> and </span>
          <a
            className="App-link"
            href="https://react-redux.js.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            React Redux
          </a>
        </span>
      </header>
    </div>
  );
}

export default App;
