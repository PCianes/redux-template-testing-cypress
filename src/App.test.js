import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from './app/store';
import App from './App';

const renderApp = () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>
  );
};

describe('<App>', () => {
  test('renders learn react link', () => {
    renderApp()

    expect(screen.getByText(/learn/i)).toBeInTheDocument();
  });

  test('renders title: "Redux template"', () => {
    renderApp()

    const title = screen.getByText(/redux template/i);

    expect(title).toBeInTheDocument();
  });

  test('title with H1', () => {
    renderApp()

    const title = screen.getByText(/redux template/i);

    expect(title.tagName).toBe('H1')
  })

  test('title with white color', () => {
    renderApp()

    const title = screen.getByText(/redux template/i);
    const style = window.getComputedStyle(title)

    expect(title).toMatchSnapshot()
    expect(style.color).toBe('rgb(255, 255, 255)')
  })

  test('render redux link to official website', () => {
    renderApp()

    const link = screen.getByText(/react redux/i)

    expect(link.tagName).toBe('A')
    expect(link.href).toBe('https://react-redux.js.org/')
  })
})

