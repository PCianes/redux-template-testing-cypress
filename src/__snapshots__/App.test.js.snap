// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> title with white color 1`] = `
<h1
  class="sc-bczRLJ bixcBr"
>
  Redux template: Jest & Cypress
</h1>
`;
